megamek/src/megamek/server/GameManager.java

```diff
@@ -11630,11 +11643,24 @@ public Vector<Report> doSetLocationsExposure(Entity entity, Hex hex,
                     entity.setLocationStatus(Mech.LOC_CLEG, ILocationExposureStatus.WET);
                     vPhaseReport.addAll(breachCheck(entity, Mech.LOC_CLEG, hex));
                 }
-            } else {
+                // Beast-mounted infantry can sit above the water level
+            } else if (!entity.isConventionalInfantry() || (entity.relHeight() < 0)) {
                 for (int loop = 0; loop < entity.locations(); loop++) {
                     entity.setLocationStatus(loop, ILocationExposureStatus.WET);
                     vPhaseReport.addAll(breachCheck(entity, loop, hex));
                 }
+            } else {
+                int status = ILocationExposureStatus.WET;
+                if (entity.isConventionalInfantry() && (entity.relHeight() >= 0)) {
+                    status = game.getPlanetaryConditions().isVacuum() ?
+                            ILocationExposureStatus.VACUUM : ILocationExposureStatus.NORMAL;
+                }
+                for (int loop = 0; loop < entity.locations(); loop++) {
```

Comment:
Is this something we actually want applied to non-infantry units?

Reply:
I was keeping the current behavior for other units, but simplifying it would fix superheavy vehicles and dropships that landing in level 1 water by putting their weapons above the waterline. A dropship that lands in depth 2+ water is destroyed, so there's no need to figure out which arcs are above the waterline in deeper water.